framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -4240,6 +4291,203 @@ private void makeGroundTargetType(
+
+        if (typeToCapture.getTypeArguments().isEmpty()
+                || (capturedTypeMirror.getTypeArguments().size()
+                        != typeToCapture.getTypeArguments().size())) {
+            return false;
+        }
+
+        for (int i = 0; i < capturedTypeMirror.getTypeArguments().size(); i++) {
+            if (TypesUtils.isCaptured(capturedTypeMirror.getTypeArguments().get(i))
+                    && typeToCapture.getTypeArguments().get(i).getKind() == TypeKind.WILDCARD) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Apply capture conversion.
+     *
+     * <p>Capture conversion is the process of converting wildcards in a generic type to fresh type
+     * variables. See <a
+     * href="https://docs.oracle.com/javase/specs/jls/se11/html/jls-5.html#jls-5.1.10">JLS
+     * 5.1.10</a> for all the details.
+     *
+     * <p>If {@code type} is not a declared type or if it does not have any wildcard type arguments,
+     * this method returns {@code type}.
+     *
+     * @param typeToCapture type to capture
+     * @return the captured type
+     */
+    public AnnotatedTypeMirror applyCaptureConversion(AnnotatedTypeMirror typeToCapture) {
+        TypeMirror capturedTypeMirror = types.capture(typeToCapture.getUnderlyingType());
+        return applyCaptureConversion(typeToCapture, capturedTypeMirror);
+    }
+
+    /**
+     * Apply capture conversion.
+     *
+     * <p>Capture conversion is the process of converting wildcards in a generic type to fresh type
+     * variables. See <a
+     * href="https://docs.oracle.com/javase/specs/jls/se11/html/jls-5.html#jls-5.1.10">JLS
+     * 5.1.10</a> for all the details.
+     *
+     * <p>If {@code type} is not a declared type or if it does not have any wildcard type arguments,
+     * this method returns {@code type}.
+     *
+     * @param type type to capture
+     * @param typeMirror type from the Java compiler that has captured wildcards, which are used as
+     *     the underlying type of the captured wildcards
+     * @return the captured type
+     */
+    public AnnotatedTypeMirror applyCaptureConversion(
+            AnnotatedTypeMirror type, TypeMirror typeMirror) {
+        if (!shouldCapture(type, typeMirror)) {
+            return type;
+        }
+
+        DeclaredType capturedTypeMirror = (DeclaredType) typeMirror;
+        AnnotatedDeclaredType typeToCapture = (AnnotatedDeclaredType) type;
+        AnnotatedDeclaredType capturedType =
+                (AnnotatedDeclaredType)
+                        AnnotatedTypeMirror.createType(capturedTypeMirror, this, false);
+        capturedType.getTypeArguments(); // initialize the type arguments
+
+        AnnotatedDeclaredType typeDeclaration =
+                (AnnotatedDeclaredType)
+                        getAnnotatedType(typeToCapture.getUnderlyingType().asElement());
+
+        // A mapping from type variable to its type argument in the captured type.
```

Comment:
The variable naming is confusing.

 * Variable `typeToCapture` refers to a single type that needs to be captured.
 * Variable `typeVarToTypeArgument` uses "To" to indicate a map type.
 * Variable `typeMirrorToCapture` also uses "To", but I'm not sure which meaning of "To" it intends.  Maybe it should be "ToCaptured"??  Another method uses the name `captureToAnnotatedCapture` for the same value (I think), which also might be better.  Once you find a good name, use it consistently.

In the following two variables, the key types are the same, but the documentation refers to it inconsistently as a "type variable" (this checks out) and a "captured type" (which seems a bit confusing).  Also, how does "the captured type" relate to the variable `capturedType`?

```java
        // A mapping from type variable to its type argument in the captured type.
        Map<TypeVariable, AnnotatedTypeMirror> typeVarToTypeArgument = new HashMap<>();
        // A mapping from the captured type to the annotated captured type.
        Map<TypeVariable, AnnotatedTypeMirror> typeMirrorToCapture = new HashMap<>();
```